### PR TITLE
Chef 15 compatibility fixes

### PR DESCRIPTION
**Pull request by detjensrobert**
Cookstyle 14 fails due to 'unnecessary cop disablement', but that disablement is necessary for Cookstyle 15.

### CODE REVIEW

**Comment by detjensrobert** [comment]
[Chef 15: osl-mysql](https://trello.com/c/YwWlv1dc/1783-chef-15-osl-mysql)

File: Berksfile [hunk 1]
```diff
@@ -3,6 +3,7 @@ source 'https://supermarket.chef.io'
 solver :ruby, :required
 
 cookbook 'apache2', '< 6.0.0'
```
Comment (ramereth) [inline Berksfile:5]: Please remove this.

File: libraries/provider_database_mysql_user.rb [hunk 2]
```diff
@@ -116,7 +110,7 @@ def ishash?
             test_sql += " AND Db='#{new_resource.database_name}'" if new_resource.database_name
             test_sql_results = test_client.query test_sql
 
-            incorrect_privs = true if test_sql_results.size.zero?
+            incorrect_privs = true if test_sql_results.size == 0 # rubocop:disable Style/ZeroLengthPredicate
```
Comment (ramereth) [inline libraries/provider_database_mysql_user.rb:113]: This actually needs to stay this way. I've run into problems with this not working using this method.

File: libraries/provider_database_mysql_user.rb [hunk 3]
```diff
@@ -338,7 +332,7 @@ def test_user_password
                           "AND authentication_string=PASSWORD('#{new_resource.password}')"
                         end
           end
-          test_client.query(test_sql).size > 0
+          !test_client.query(test_sql).size == 0 # rubocop:disable Style/ZeroLengthPredicate
```
Comment (ramereth) [inline libraries/provider_database_mysql_user.rb:335]: This needs to stay this way as I ran into issues trying this before.

File: libraries/provider_database_mysql_user.rb [hunk 4]
```diff
@@ -369,7 +363,7 @@ def update_user_password
         end
 
         def database_has_password_column(client)
-          client.query('SHOW COLUMNS FROM mysql.user WHERE Field="Password"').size > 0
+          !client.query('SHOW COLUMNS FROM mysql.user WHERE Field="Password"').size == 0 # rubocop:disable Style/ZeroLengthPredicate
```
Comment (ramereth) [inline libraries/provider_database_mysql_user.rb:366]: This needs to stay this way as I ran into issues trying this before.

File: spec/mon_spec.rb [hunk 5]
```diff
@@ -6,7 +6,7 @@
   ALLPLATFORMS.each do |pltfrm|
     context "on #{pltfrm[:platform]} #{pltfrm[:version]}" do
       cached(:chef_run) do
-        ChefSpec::SoloRunner.new(pltfrm).converge(described_recipe)
+        ChefSpec::SoloRunner.new(pltfrm.dup.merge(step_into: %w(munin_plugin))).converge(described_recipe)
```
Comment (ramereth) [inline spec/mon_spec.rb:9]: We shouldn't be stepping into the ``munin_plugin`` resource and instead should only be checking the munin_plugin resource in ChefSpec.

File: spec/mon_spec.rb [hunk 6]
```diff
@@ -43,24 +43,11 @@
...
-            connection: {
-              host: 'localhost',
-              username: 'root',
-              password: 'root_pw',
-            },
-            username: 'monitor',
-            password: 'monitor_pw',
-            privileges: [:super, :select, :process, 'replication client', 'replication slave']
-          )
-      end
```
Comment (ramereth) [inline spec/mon_spec.rb:58]: Why was this removed?

File: libraries/resource_database_user.rb [hunk 7]
```diff
@@ -34,81 +38,18 @@ def initialize(name, run_context = nil)
...
-        )
-      end
-
-      def require_x509(arg = nil)
-        set_or_return(
-          :require_x509,
-          arg,
-          kind_of: [TrueClass, FalseClass]
-        )
+#        @allowed_actions.push(:create, :drop, :grant, :revoke)
```
Comment (ramereth) [inline libraries/resource_database_user.rb:41]: Why is this commented out?

File: Berksfile (hunk 1 above)
Comment (ramereth) [reply Berksfile]: So actually it looks like we should keep this but update it to be ``< 8.2`` As 8.2 and above require Chef 15 which breaks ChefSpec on 14.

**Comment by detjensrobert** [comment]
!bump minor *

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a minor-level version bump to v2.4.0.  Have a nice day!
